### DISCUSSION TITLE
Change store default suffix to: '', rather than: 'Store'; for better code comprehension and search

### DISCUSSION THREAD

**Discussion by jimattakeda** [vuejs/pinia#2018] [Ideas]
After many years of programming, in a diverse variety of languages, I have come to the firm conclusion that _dynamically constructing identifiers_ (in a program) is a bad idea. My justification for this stance is:

1. When an identifier is dynamically constructed, external/traditional string search tools (e.g., `grep`) cannot be used to find any of the references to a given identifier.
2. Likewise, IDE code search tools also fail to 'match' and to find those dynamically constructed identifiers. (Though Regexes could help, but only if the Programmer is already familiar with Pinia, in detail.)
3. Having something that looks like a simple string property (e.g., `id: 'myStore'`) get mangled [1] into something else (presumably: `"myStoreStore"`, in this case, is: confusing, in general, and is particularly problematic for programmers who are un-familiar with Pinia (i.e., new team members, or a programmer who is unfamiliar with a project, but needs to do emergency technical support on that project).
4. Having identifiers get mangled (by an obscure process) arguably contributes to one of the two hardest things in computer science: _naming things_ [2].
5. Six key-strokes (the five letters of "store" plus the shift key) is NOT too much to type--code 'golfing' is an ANTI-pattern, IMO.
6. It violates the 'principle of least surprise'.

**My proposal: change the Pinia store creation/access logic so that the default suffix is just an empty string: ''**. I realize that this would be a breaking change, for existing code, so this change might need to go through a typical: announcement, deprecation, actual change, cycle. And that would take longer to happen. But I think it is still well worth it.

Before someone mentions it: yes, I am aware of the existing Pinia `setMapStoreSuffix()` function [3]. However, I do not consider that facility an acceptable substitute for what this proposal is asking for. It looks like a leaky abstraction, combining: TypeScript interfaces, redundant values (e.g., ''foo' for the suffix setting, and 'foo' for the interface; module declaration, and (possibly) ensuring that the code flow is in a certain order, i.e., `setMapStoreSuffix()` --> `createPinia()` --> `declare module 'pinia'`. 

Also, as additional, personal, personal motivation for this proposal, I have just spent several hours trying to get `setMapStoreSuffix()` to do what I want--but to no avail (and, NO, this is NOT a request for debugging assistance, it IS an experience report from a very experienced programmer).

Further, _constructive_ discussion is welcome.

[1] I'm using "mangled" in a technical sense here (e.g., C++ name mangling); and NOT a pejorative sense.
[2] https://www.karlton.org/2017/12/naming-things-hard/
[3] https://pinia.vuejs.org/cookbook/options-api.html#usage-without-setup

**Comment by posva** [comment]
Sorry for teh late reply. This could create too many conflicts with other global properties. IT's also one of the reasons the setup api is preferred over the options api and why it's a configurable option so more advanced users can change it to match their own conventions.
